WordPress/gutenberg
# Migrate post editor feature preferences to the interface package

## Description
Partly addresses #31965. Related to #31965 and #34135.

Migrates post editor feature preferences over to the interface package.

Here there was a choice of deprecating the post editor's APIs for features, but this keeps them as backwards compatibility is pretty easy to maintain. Any selectors and actions call through to the interface store's implementation.

## How has this been tested?
1. Checkout trunk and open the post editor.
2. Set some preferences to a value different to the default (e.g. 'Top toolbar' and 'Contain cursor inside block' to active.)
3. Checkout this branch, rebuild and open up the post editor
4. The welcome guide should not show and the preferences should be retained from step 2.
5. Clear browser storage (or just the WP_DATA_USER_ values) and reload
6. Open up the customize widgets editor, the welcome guide should be shown
7. Try toggling various settings to ensure they work.

## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
